native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBResultParameterProcessor.java

```diff
@@ -132,4 +139,125 @@ public BObject getBalStreamResultIterator() {
+                                                 ColumnDefinition columnDefinition)
+            throws ApplicationError, SQLException {
+        int sqlType = columnDefinition.getSqlType();
+        Type ballerinaType = columnDefinition.getBallerinaType();
+        switch (sqlType) {
+            case -104:
+            case -103:
+                return processIntervalResult(resultSet, columnIndex, sqlType, ballerinaType,
+                        columnDefinition.getSqlName());
+            case -101:
+            case -102:
+                return processTimestampWithTimezoneResult(resultSet, columnIndex, sqlType, ballerinaType);
+            default:
+                throw new ApplicationError("Unsupported SQL type " + columnDefinition.getSqlName());
+        }
+    }
+
+    @Override
+    public Object convertTimeStamp(java.util.Date timestamp, int sqlType, Type type) throws ApplicationError {
+        Utils.validatedInvalidFieldAssignment(sqlType, type, "SQL Date/Time");
+        if (timestamp != null) {
+            switch (type.getTag()) {
+                case TypeTags.STRING_TAG:
+                    return fromString(timestamp.toString());
+                case TypeTags.OBJECT_TYPE_TAG:
+                case TypeTags.RECORD_TYPE_TAG:
+                    if (type.getName().equalsIgnoreCase(io.ballerina.stdlib.time.util.Constants.CIVIL_RECORD)
+                            && timestamp instanceof Timestamp) {
+                        return Utils.createTimestampRecord((Timestamp) timestamp);
+                    } else if (type.getName().equalsIgnoreCase(io.ballerina.stdlib.time.util.Constants.DATE_RECORD)
+                            && timestamp instanceof Timestamp) {
+                        return Utils.createDateRecord(new java.sql.Date(timestamp.getTime()));
+                    } else {
+                        throw new ApplicationError("Unsupported Ballerina type:" +
+                                type.getName() + " for SQL Timestamp data type.");
+                    }
+                case TypeTags.INT_TAG:
+                    return timestamp.getTime();
+                case TypeTags.INTERSECTION_TAG:
+                    return Utils.createTimeStruct(timestamp.getTime());
+            }
+        }
+        return null;
+    }
+
+    private Object processIntervalResult(ResultSet resultSet, int columnIndex, int sqlType, Type ballerinaType,
+                                         String sqlTypeName) throws ApplicationError, SQLException {
+        String intervalString = resultSet.getString(columnIndex);
+        return convertInterval(intervalString, sqlType, ballerinaType, sqlTypeName);
+    }
+
+    private Object convertInterval(String interval, int sqlType, Type ballerinaType, String sqlTypeName)
+            throws ApplicationError {
+        if (interval != null) {
+            switch (ballerinaType.getTag()) {
+                case TypeTags.STRING_TAG:
+                    return  fromString(interval);
```

Comment:
```suggestion
                    return fromString(interval);
```

Reply:
fixed in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/13/commits/950f2228fe9fde68f2865688c03597cabbea3792